### PR TITLE
Fix warnings and errors for latest Swift 2.0 (Xcode 7 beta 6).

### PR DESCRIPTION
**Pull request by khanov**

### CODE REVIEW

File: CryptoSwift/MD5.swift [hunk 1]
```diff
@@ -107,11 +107,11 @@ final class MD5 : HashProtocol  {
             hh[3] = hh[3] &+ D
         }
 
-        let buf: NSMutableData = NSMutableData();
-        hh.map({ (item) -> () in
+        let buf: NSMutableData = NSMutableData()
+        hh.forEach { item in
```
Comment (krzyzanowskim) [inline CryptoSwift/MD5.swift]: map has been changed to forEach by the converter ?

Comment (khanov) [inline CryptoSwift/MD5.swift]: Nope, I changed it myself. There was a warning that the result of map wasn't used.

File: CryptoSwift/ChaCha20.swift [hunk 2]
```diff
@@ -194,7 +194,7 @@ final public class ChaCha20 {
 /// Change array to number. It's here because arrayOfBytes is too slow
 private func wordNumber(bytes:ArraySlice<UInt8>) -> UInt32 {
     var value:UInt32 = 0
-    for (var i:UInt32 = 0, j = 0; i < 4; i++, j++) {
+    for (var i:UInt32 = 0, j = Int(bytes.indices.startIndex); i < 4; i++, j++) {
```
Comment (krzyzanowskim) [inline CryptoSwift/ChaCha20.swift]: this change should not be needed

**Comment by krzyzanowskim** [comment]
@khanov I merged the other one PR, if you want, please update this changes to the current codebase

**Pull request closed without merging**